apis/resourcecontrollerv2/v1alpha1/resourcekey_types.go

```diff
@@ -103,19 +102,18 @@ type ResourceKeyObservation struct {
 	DeletedAt *metav1.Time `json:"deletedAt,omitempty"`
 
 	// The subject who created the key.
-	CreatedBy string `json:"created_by,omitempty"`
+	CreatedBy string `json:"createdBy,omitempty"`
 
 	// The subject who updated the key.
-	UpdatedBy string `json:"updated_by,omitempty"`
+	UpdatedBy string `json:"updatedBy,omitempty"`
 
 	// The subject who deleted the key.
-	DeletedBy string `json:"deleted_by,omitempty"`
+	DeletedBy string `json:"deletedBy,omitempty"`
 }
 
 // A ResourceKeySpec defines the desired state of a ResourceKey.
 type ResourceKeySpec struct {
 	runtimev1alpha1.ResourceSpec `json:",inline"`
-	ConnectionTemplates          map[string]string     `json:"connectionTemplates,omitempty"`
```

Comment:
Can you explain why this is here, so I can add the proper logic to generate when appropriate?

Reply:
yes, this is used whenever we have confidential data that goes into a secret. It allows to use go templating to extract the keys that are set into the secret.The code to handle these templates is pretty much a common utility: 
https://github.com/pdettori/provider-ibm-cloud/blob/rcv2-key/pkg/clients/template_parser.go
and it is used here: https://github.com/pdettori/provider-ibm-cloud/blob/rcv2-key/pkg/clients/resourcekey/resourcekey.go#L128-L173
An example of how this is used is here:
https://github.com/pdettori/provider-ibm-cloud/blob/rcv2-key/examples/resourcecontrollerv2/resource-key-postgres.yaml